File: 프로그래머스/lv0/120852. 소인수분해/소인수분해.js
function solution(n) {
    var answer = new Set();
    while(n > 1) {
        for(var i=2; i<=n; i++){
            if(n%i === 0){
                answer.add(i);
                n/=i;
                break;
            }
        }
    }
    return [...answer];
}

File: 프로그래머스/lv0/120852. 소인수분해/소인수분해.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        Set<Integer> set = new HashSet<>();
        int[] answer = {};
        for(int i=2; i<=n; i++){
            while(n%i ==0){
                set.add(i);
                n/=i;
            }
        }
        answer = set.stream().mapToInt(Integer::intValue).toArray();;
        Arrays.sort(answer);
        return answer;
    }
}